TypeCobol/Compiler/CodeElements/Entry/DataType.cs

```diff
@@ -23,18 +24,37 @@ public class DataType : ICobolLanguageLevel, IVisitable {
 
 		public override string ToString() { return Name; }
 
-		public override int GetHashCode() { return Name.GetHashCode(); }
 	    public bool AcceptASTVisitor(IASTVisitor astVisitor) {
 	        return astVisitor.Visit(this);
 	    }
 
 	    public override bool Equals(object obj) {
-			var other = obj as DataType;
-			if (other == null) return false;
-			return other == this;
-		}
+            return Equals(obj as DataType);
+        }
+
+        public bool Equals(DataType dataType)
+        {
+            if (Object.ReferenceEquals(this, dataType)) return true;
+            if (Object.ReferenceEquals(null, dataType)) return false;
+
+            //Data instance for Cobol85 are unique so their references have been compared
+            if (CobolLanguageLevel == CobolLanguageLevel.Cobol85 || dataType.CobolLanguageLevel == CobolLanguageLevel.Cobol85) return false;
+
+            return Name.Equals(dataType.Name, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = 13;
+                hashCode = (hashCode * 397) ^ Name.GetHashCode();
+
+                return hashCode;
+            }
+        }
```

Comment:
CobolLanguageLevel and RestrictionLevel should be used in comparison along with Name.

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/pull/1600/commits/8b7686fa9f02c00e2a7150589c65d7fd66b787c8